src/software/amazon/ion/IonInt.java

```diff
@@ -56,6 +56,15 @@ public long longValue()
     public BigInteger bigIntegerValue();
 
     /**
+     * Gets the actual type of the underlying content.
+     *
+     * @return an {@link IntegerType} denoting the content's Java type,
+     * or <code>null</code> if this is <code>null.int</code>.
```

Comment:
Should this return reference null when we have IntegerType.NONE?

I think we either have reference null or the NONE value, but not both.